src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -619,6 +619,12 @@ public bool IsActivityPointer
                     return _isActivityPointer.Value;
                 }
 
+                // TryGetType assumes Kind is valid
+                if (Kind == DKind.Invalid)
+                {
+                    return false;
+                }
```

Comment:
are we confident this won't cause any regressions?

Reply:
this change isn't needed - I just found it when trying to Yaml-serialize some weird object by mistake..

I don't think there is any risk as TryGetType starts with AssertValid()

Reply:
Contracts.Assert don't run in retail. 
So it's possible that PA or another client is actually hitting that assert, ignoring it, and getting a result. And the check here could change it. 

Just a potential risk - it's safest to just leave the file alone - because as soon as we even "touch" the file, this PR becomes suspect on future binding failures.

Reply:
Removed